client/session.go

```diff
@@ -1521,6 +1521,10 @@ func (s *session) readConsistencyResult(
 	return nil
 }
 
+func (s *session) IteratorPools() (IteratorPools, error) {
+	return s.pools, nil
```

Comment:
Might want to check that `s.pools != nil`?  Also are the pools always guaranteed to be available (I can't remember when they get alloc'd).  If not we probably have to nil check the specific pools in question too =[

Reply:
s.pools is a `sessionPools` struct, and isn't itself nilable.

For the pools themselves, I believe they get alloced in `session.Open()`, so I think it should be safe unless you get the `session` in some weird way. Adding a sanity check would make sense, though.

Reply:
maybe ensure the session is open then?

```
	s.state.RLock()
        defer s.state.RULock()
	if s.state.status != statusOpen {
		return nil, errSessionStatusNotOpen
	}
        return s.pools
```